docs/guide/authentication.md

```diff
@@ -1,5 +1,117 @@
+Encoded into the certificate's `Common Name` field is the user name. The
+certificate's `organization` fields indicate the user's OpenStack role
+assignments.
+
+This effectively maps OpenStack roles to Kubernetes groups.
+
+These certificates are generated. They can be retrieved via UI or API. In order
+to allow for revocation of authorization the certificates are short lived. They
+automatically expire after 24h. Therefore they need to be periodically
+refreshed.
+
+### Authorizations
+
+Using the `user` and `groups` provided by the authentication mechanism it is
+then possible to use [Kubernetes
+RBAC](https://kubernetes.io/docs/admin/authorization/rbac/) to define
+authorizations within Kubernetes.
+
+By distinguishing between users as well as `kubernikus_admin` and `kubernikus_member`
+roles/groups it is possible to assign different Kubernetes roles to groups or
+individual users.
+
+## Manage Roles in OpenStack
+
+Users with `Keystone Administrator` role are allowed to change user role
+assignments in a project.
+
+![User Role
+Assignments](/assets/images/docs/containers/kubernetes/userroleassignments.png)
+
+To add additional users to a cluster they need to be
+given either `Kubernetes Admin` or `Kubernetes Member` roles.
+
+![Role Assignments](/assets/images/docs/containers/kubernetes/roleassignment.png)
+/assets/images/docs/containers/kubernetes/userroleassignments.png
+
+## Authenticating with Kubernetes
+
+Kubernetes is usually remote controlled with the `kubectl` command line tool.
+It is configured through a config file `.kubeconfig`. For installation
+instructions please see the [official
+documentation](https://kubernetes.io/docs/user-guide/kubectl-overview/).
+
+### Manual Download
+A preconfigured `.kubeconfig` file can be downloaded from the UI or fetched via
+API:
+
+![Download Credentials](/assets/images/docs/containers/kubernetes/credentials.png)
+
+### Automatic Refresh
+
+Since the certificates expire daily it becomes quite tedious to download new
+`.kubeconfig` files every day. To help with this workflow there is a CLI tool
+`kubernikusctl` for remote controlling Kubernikus clusters.
+
+The tool is available precompiled for Linux, MacOS and Windows. The latest
+version can be downloaded from [Github](https://github.com/sapcc/kubernikus/releases/latest).
+
+Setting up an automatic refresh of the `.kubeconfig` file is a 2-step process:
+
+  1. `kubernikusctl auth init --many --auth --options`
+  2. `kubernikusctl auth refresh`
+
+The initialisation only needs to be done once. Afterwards a `refresh` is
+possible without repeating all authentication details.
+
+![Setup](/assets/images/docs/containers/kubernetes/setup.png)
+
+The UI provides the full `kubernikusctl auth init` initialisation command for
+convenience.
+
+### Default Permissions
+
+By default any user with the `Kubernetes Admin` OpenStack role is assigned the
+`cluster-admin` Kubernetes role. This is a super-admin that is allowed
+everything.
```

Comment:
Maybe we should show the concrete RBAC rule we preseed with every cluster here.

Reply:
+1 See https://github.com/sapcc/kubernikus/issues/122

We have a seed for the `kubernetes_admin` already. Should be easy to extent.